packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -647,3 +648,20 @@ export function didNotFindHydratableTextInstance(
     warnForInsertedHydratedText(parentInstance, text);
   }
 }
+
+// in a dom-like test environment, we want to warn if dispatchAction()
+// is called outside of a batchedUpdates/TestUtils.act(...) call.
+let ensureBatchedDispatchedActions = false;
+
+if (__DEV__) {
+  if (
+    canUseDOM &&
+    (navigator.userAgent.indexOf('Node.js') >= 0 ||
+      navigator.userAgent.indexOf('jsdom') >= 0)
```

Comment:
This _might_ be set to something else (https://jestjs.io/docs/en/configuration#testenvironmentoptions-object).

Might be a good idea to also check for some sign it's Jest (e.g. Jest can start to do `global.thisIsCatuallyJest = Symobl.for('jest-yes-this-is-jest')` or something). I saw the earlier talk about checking for `global.it` - at some point Jest will want to provide a global-less API so it'd be great to not couple anything more to it (although not in the immediate future)

Reply:
Changed it so that it only detects jest now.

Reply:
that might also go away at some point, but not in the foreseeable future. Thanks!